src/Core/Domain/Customer/Group/Command/AddCustomerGroupCommand.php

```diff
@@ -75,4 +112,40 @@ public function showPrice(): bool
     {
         return $this->showPrice;
     }
+
+    /**
+     * @return ShopId[]
+     */
+    public function getShopIds(): array
+    {
+        return $this->shopIds;
+    }
+
+    private function assertIsCorrect()
+    {
+        $this->assertReductionIsEqualOrGreaterThanZero();
+    }
+
+    private function assertReductionIsEqualOrGreaterThanZero()
+    {
+        $this->assertIsPrice($this->reduction);
+    }
+
+    private function assertIsPrice(DecimalNumber $price)
+    {
+        $regexCheck = preg_match('/^[0-9]{1,10}(\.[0-9]{1,9})?$/', (string) $price);
+        if (0 === $regexCheck || false === $regexCheck) {
+            throw new InvalidReductionException();
+        }
```

Comment:
I don't think this check is logical, you already have a DecimalNumber here, you don't need to convert it to string and check if it was converted correctly. It just doesn't make sense.

Reply:
Better naming, took the regex from legacy but it's not needed because of DecimalNumber you were right 👍